plugins/modules/packaging/os/snap.py

```diff
@@ -105,151 +105,184 @@
+        notes = match.group('notes')
+        return "disabled" not in notes.split(',')
+
+    def validate_input_snaps(self):
+        """Ensure that all exist."""
+        for snap_name in self.vars.name:
+            if not self.snap_exists(snap_name):
+                raise ModuleHelperException(msg="No snap matching '%s' available." % snap_name)
+
+    def state_present(self):
+        self.validate_input_snaps()  # if snap doesnt exist, it will explode when trying to install
+        self.vars.meta('classic').set(output=True)
+        self.vars.meta('channel').set(output=True)
+        actionable_snaps = [s for s in self.vars.name if self.is_snap_installed(s)]
+        if not actionable_snaps:
+            return
+        self.changed = True
+        self.vars.snaps_installed = actionable_snaps
+        if self.module.check_mode:
+            return
+        params = ['classic', 'channel', 'state']  # get base cmd parts
+        has_one_pkg_params = bool(self.vars.classic) or self.vars.channel != 'stable'
+        has_multiple_snaps = len(actionable_snaps) > 1
+        if has_one_pkg_params and has_multiple_snaps:
+            commands = [params + [s] for s in actionable_snaps]
+        else:
+            commands = [params + actionable_snaps]
+        self.vars.cmd, rc, out, err = self._run_multiple_commands(commands)
+        if rc == 0:
+            return
+
+        classic_snap_pattern = re.compile(r'^error: This revision of snap "(?P<package_name>\w+)"'
+                                          r' was published using classic confinement')
+        match = classic_snap_pattern.match(err)
+        if match:
+            err_pkg = match.group('package_name')
+            msg = "Couldn't install {name} because it requires classic confinement".format(name=err_pkg)
+        else:
+            msg = "Ooops! Snap installation failed while executing '{cmd}', please examine logs and " \
+                  "error output for more details.".format(cmd=self.vars.cmd)
+        raise ModuleHelperException(msg=msg)
+
+    def state_absent(self):
+        self.validate_input_snaps()  # if snap doesnt exist, it will be absent by definition
+        actionable_snaps = [s for s in self.vars.name if not self.is_snap_installed(s)]
+        if not actionable_snaps:
+            return
+        self.changed = True
+        self.vars.snaps_removed = actionable_snaps
+        if self.module.check_mode:
+            return
+        params = ['classic', 'channel', 'state']  # get base cmd parts
+        commands = [params + actionable_snaps]
+        self.vars.cmd, rc, out, err = self._run_multiple_commands(commands)
+        if rc == 0:
+            return
+        msg = "Ooops! Snap removal failed while executing '{cmd}', please examine logs and " \
+              "error output for more details.".format(cmd=self.vars.cmd)
+        raise ModuleHelperException(msg=msg)
```

Comment:
Are the logs automatically added to the output? That's not clear to me here. (Same in similar places.)

Reply:
That message was not changed from the previous version, and I understood *logs* in here as referring to the `ansible-playbook` output. Maybe we could make that clearer.

Reply:
@felixfontein Doest that answer your question? Is there anything else we are waiting for in this one?

Reply:
I understand logs as output of `cmd` that was executed. Because that output does not seem to be included anywhere.

Reply:
Well, with the major revamp (using MH), the underlying command's outcome (rc, out, err) always gets added to the module results.